### PR TITLE
Remove not longer needed functions

### LINKED ISSUE #26244: OCShare20 Provider getChildren method not needed
According to https://github.com/owncloud/core/blob/stable9.1/lib/private/Share20/Manager.php#L741
the provider getChildren method should not be needed.

We are implementing our own sharing and we would like to reuse your Share Manager in the cleanest possible way, but that method stop us as we do not want to have such method in our provider.

The blocker PR has been merged so I see no need to have it there.

The workaround we have now is this:
https://github.com/owncloud/core/compare/stable9.1...cernbox:neweosstorage?expand=1#diff-e19e214fd7a3f2f268ba743b7513a475

https://github.com/owncloud/core/compare/stable9.1...cernbox:neweosstorage?expand=1#diff-a9d8fe189a37b652ea5b821e4d772438

Cheers

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description

Methods are no longer required
## Related Issue

fixes #26244
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Why? 

```
12:29:09 1) OCA\Files_Sharing\Tests\ApiTest::testInvisibleSharesUser
12:29:09 Failed asserting that an array is empty.
12:29:09 
12:29:09 /var/lib/jenkins/workspace/owncloud-core_core_PR-26335-I2EL7QZH3LTPC5RFIC4IQWFXFYI5QCULR6S3JUQNQHD6D6IICTJQ/apps/files_sharing/tests/ApiTest.php:1596
12:29:09

**Comment by PVince81** [comment]
Hmm no idea... I hope those link shares weren't still some kind of children.

**Comment by PVince81** [comment]
See https://github.com/owncloud/core/issues/22295 which is linked in the test PHPDoc

**Comment by PVince81** [comment]
moving to backlog

to be rescheduled as part of tech debt handling

File: lib/private/Share20/DefaultShareProvider.php [hunk 1]
```diff
@@ -877,6 +839,7 @@ private function createShare($data) {
 	 * - $chunkedShareIds responsible to split shareIds into chunks containing 100 elements
 	 * 	 e.g. $chunkedShareIds { { "4", "52", "54",... }[100], { .. }[2] }[2]
 	 *
+<<<<<<< HEAD
```
Comment (PVince81) [inline lib/private/Share20/DefaultShareProvider.php]: ohhhh

Comment (DeepDiver1975) [inline lib/private/Share20/DefaultShareProvider.php]: one rebase to destroy them all .....

**Comment by PVince81** [comment]
I ran the debugger on this and the array **IS** empty, wtf

**Comment by PVince81** [comment]
oops, wrong test

**Comment by PVince81** [comment]
ok, so this is the share that is wrongly returned:
```
    $result->data[0]             = (array[25]);
      $result->data[0]['id']     = (string[2]) '53';
      $result->data[0]['share_type'] = (int) '3';
      $result->data[0]['uid_owner'] = (string[16]) 'test-share-user2';
      $result->data[0]['displayname_owner'] = (string[16]) 'test-share-user2';
      $result->data[0]['permissions'] = (int) '1';
      $result->data[0]['stime']  = (int) '1499258689';
      $result->data[0]['parent'] = (null);
      $result->data[0]['expiration'] = (null);
      $result->data[0]['token']  = (string[15]) 'jwq0Rgy0q1qGFra';
      $result->data[0]['uid_file_owner'] = (string[16]) 'test-share-user1';
      $result->data[0]['displayname_file_owner'] = (string[16]) 'test-share-user1';
      $result->data[0]['path']   = (string[47]) '/folder_share_api_test/subfolder_share_api_test';
      $result->data[0]['item_type'] = (string[6]) 'folder';
      $result->data[0]['mimetype'] = (string[20]) 'httpd/unix-directory';
      $result->data[0]['storage_id'] = (string[22]) 'home::test-share-user1';
      $result->data[0]['storage'] = (int) '57';
      $result->data[0]['item_source'] = (int) '446';
      $result->data[0]['file_source'] = (int) '446';
      $result->data[0]['file_parent'] = (int) '445';
      $result->data[0]['file_target'] = (string[25]) '/subfolder_share_api_test';
      $result->data[0]['share_with'] = (null);
      $result->data[0]['share_with_displayname'] = (null);
      $result->data[0]['name']   = (null);
      $result->data[0]['url']    = (string[44]) 'http://localhost/index.php/s/jwq0Rgy0q1qGFra';
      $result->data[0]['mail_send'] = (int) '0';
```

**Comment by PVince81** [comment]
and here about "invisible shares": https://github.com/owncloud/core/issues/22295

**Comment by PVince81** [comment]
I've just tested the web UI and the invisible links do **not** appear. So the unit test result doesn't match what happens in the other layers.

I still don't why this one is even affected by your changes.

File: lib/private/Share20/Manager.php [hunk 2]
```diff
@@ -809,9 +786,6 @@ public function deleteShare(\OCP\Share\IShare $share) {
 		// Emit pre-hook
 		\OC_Hook::emit('OCP\Share', 'pre_unshare', $hookParams);
 
-		// Get all children and delete them as well
-		$deletedShares = $this->deleteChildren($share);
```
Comment (PVince81) [inline lib/private/Share20/Manager.php]: if I add this line back (with the functions), the failing test passes... so we do still need that function after all...

I see this is in `deleteShare`, so it's likely that when "user1" unshares from "user2", we currently (on master) also delete all child shares created by "user2".

**Comment by PVince81** [comment]
My research shows that we cannot remove `deleteChildren` because it is still needed when deleting shares in the following scenario:

1. user1 shares foo with user2
1. user2 shares foo by link
1. user1 unshares from user2

At this point user2 has no more access to "foo" but oc_share still contains the link share on this PR.
Running cron doesn't seem to clean up this kind of orphaned shares...

We could improve the repair steps for that though.

**Comment by PVince81** [comment]
still needed, will need to be revisited at some point in the future

**Pull request closed without merging**